addons/maintainence-agent/maintenance_mode_controller.go

```diff
@@ -122,7 +138,7 @@ func (r *MaintenanceModeReconciler) startMaintenanceActions(ctx context.Context,
 				klog.Info("Scaling up RBD mirror deployments")
 				replicas = 1
 			}
-			_, err := r.scaleRBDMirrorDeployment(ctx, clusters, replicas)
+			_, err := r.scaleRBDMirrorDeployment(ctx, *cluster, replicas)
```

Comment:
Status updates after this line can probably be handled better.
returning `err` is more important that returning `statusErr`.

Reply:
Changed the func to setStatus since we are updating at the end of reconcile